File: abc016/src/b.go
package main

import "fmt"

func main() {
	var a, b, c int
	fmt.Scan(&a, &b, &c)
	p := a+b == c
	m := a-b == c
	var ans byte
	if p && m {
		ans = '?'
	} else if p && !m {
		ans = '+'
	} else if !p && m {
		ans = '-'
	} else {
		ans = '!'
	}
	fmt.Printf("%c\n", ans)
}

File: abc016/src/b.py
a, b, c = map(int, input().split())
p = a + b == c
m = a - b == c
if p and m:
    ans = '?'
elif p and not m:
    ans = '+'
elif not p and m:
    ans = '-'
else:
    ans = '!'
print(ans)